scapy/contrib/nlm.py

```diff
@@ -0,0 +1,307 @@
+# This file is part of Scapy
+# See http://www.secdev.org/projects/scapy for more information
+# Copyright (C) Lucas Preston <lucas.preston@infinite.io>
+# This program is published under a GPLv2 license
+
+# scapy.contrib.description = Network Lock Manager (NLM) v4
+# scapy.contrib.status = loads
+
+import scapy.contrib.oncrpc as rpc
+from binascii import unhexlify
+from scapy.packet import Packet, bind_layers
+from scapy.fields import IntField, StrLenField, LongField, PacketField, \
+    IntEnumField
+
+nlm4_stats = {
+    0: 'NLM4_GRANTED',
+    1: 'NLM4_DENIED',
+    2: 'NLM4_DENIED_NOLOCKS',
+    3: 'NLM4_BLOCKED',
+    4: 'NLM4_DENIED_GRACE_PERIOD',
+    5: 'NLM4_DEADLCK',
+    6: 'NLM4_ROFS',
+    7: 'NLM4_STALE_FH',
+    8: 'NLM4_FBIG',
+    9: 'NLM4_FAILED'
+}
+
+
+class File_Object(Packet):
+    name = 'File Object'
+    fields_desc = [
+        IntField('length', 0),
+        StrLenField('fh', b'', length_from=lambda pkt: pkt.length),
+        StrLenField('fill', b'', length_from=lambda pkt: (4 - pkt.length) % 4)
+    ]
+
+    def set(self, new_filehandle, length=None, fill=None):
+        # convert filehandle to bytes if it was passed as a string
+        if isinstance(new_filehandle, str):
+            new_filehandle = unhexlify(new_filehandle)
```

Comment:
This won't have the same effect in a Python 2 :/

However you could use `not isinstance(..., bytes):` so that the case works in Py3, and is ignored in Py2 (the user will have to pass valid bytes in Py2)

Reply:
Addressed by `1953ff47`. Changed the check to `new_filehandle.isalnum()` which removes an `unhexlify` Exception if a (non-bytes) string like `\x00\x00\x00\x00` is passed